### PR TITLE
NIFI-3145 Rewriting double validation in NumberParsing

### PR DESCRIPTION
**Pull request by JPercivall**
Adding more tests to TestQuery

### CODE REVIEW

**Comment by mattyb149** [comment]
Reviewing...

File: nifi-commons/nifi-expression-language/src/main/java/org/apache/nifi/attribute/expression/language/evaluation/util/NumberParsing.java [hunk 1]
```diff
@@ -21,54 +21,57 @@
...
-                    ")[pP][+-]?" + Digits + "))" +
-                    "[fFdD]?))" +
-                    "[\\x00-\\x20]*");// Optional trailing "whitespace"
-
-    private static final Pattern DOUBLE_PATTERN = Pattern.compile(fpRegex);
-
-    private static final Pattern NUMBER_PATTERN = Pattern.compile("-?((\\d+)|(0[xX]" + HexDigits + "))");
+    private static final String OptionalSign  = "[\\-\\+]?";
+
+    private static final String Infinity = "(Infinity)";
```
Comment (mattyb149) [inline nifi-commons/nifi-expression-language/src/main/java/org/apache/nifi/attribute/expression/language/evaluation/util/NumberParsing.java]: Should the parenthese be around Infinity and NaN? If you perform literal(0):toDecimal():divide(0.0) you get "NaN" without parens, and literal("-3"):toDecimal():divide(0.0) you get "-Infinity" without parens. If you do literal("NaN"):toDecimal():plus(1) you get "NaN", and if you use literal("(NaN)").toDecimal():plus(1) you get an empty value.

Comment (mattyb149) [inline nifi-commons/nifi-expression-language/src/main/java/org/apache/nifi/attribute/expression/language/evaluation/util/NumberParsing.java]: Ah, I see it's only for the regex, not an actual token. I take that back :)  However in the future we may want to consider using the lexer/parser classes (invoking a lower-level rule) so we don't have multiple ways of recognizing the elements.

**Comment by JPercivall** [comment]
@mattyb149 I just pushed out a new commit adding the "a-f" handling and tests for it.

**Comment by mattyb149** [comment]
+1 LGTM, tested various double representations and operations. Thank you! Merging to master

**Pull request closed without merging**